File: leetcode/contest-257/1996. The Number of Weak Characters in the Game.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
   public:
    int numberOfWeakCharacters(vector<vector<int>>& properties) {
        sort(properties.begin(), properties.end());
        priority_queue<pair<int, int>> pq;
        int n = properties.size();
        int ans = 0;
        for (int i = n - 1; i >= 0;) {
            int a = properties[i][0];
            int j = i;
            for (; j >= 0; j--) {
                if (properties[j][0] != a) break;
                if (pq.empty())
                    continue;
                else if (pq.top().first > properties[j][1])
                    ans++;
            }
            for (int k = i; k > j; k--) {
                pq.push({properties[k][1], properties[k][0]});
            }
            i = j;
        }
        return ans;
    }
};

File: leetcode/残酷刷题/1996. The Number of Weak Characters in the Game.ts
function numberOfWeakCharacters (properties: number[][]): number {
  properties.sort((a, b) => a[0] - b[0])
  const n = properties.length
  let max = 0
  let ans = 0
  for (let i = n - 1; i >= 0;) {
    const curX = properties[i][0]
    let j = i; const lastMax = max
    for (; j >= 0; j--) {
      const [x, y] = properties[j]
      if (x !== curX) break
      if (lastMax > y) ans++
      max = Math.max(max, properties[j][1])
    }
    i = j
  }
  return ans
};
